### ISSUE TITLE
assert fails in HwRenderPrimitiveFactory ctor

### ISSUE THREAD

**Issue by teub** [google/filament#5724]
**Describe the bug**
Updating to the latest version of filament (1.23.2), I get a failed assert in in HwRenderPrimitiveFactory() in debug and a crash later in release which I suspect is related to that assert.

**To Reproduce**
Just create the engine with filament::Engine::create(filament::Engine::Backend::OPENGL).
This is possibly a problem specific to visual studio ? See the notes below.

**Desktop (please complete the following information):**
 - OS: Windows, VS 2022
 - Backend: OpenGL

**Additional context**

The assert that fails is in debug, and I have a crash later in release which I suspect is related to this assert.
See at the bottom for the callstack of the assert.
I suspect this is related to commit 7aa3b1a6eef7749059541d904ae0993daa445263 "merge identical backend RenderPrimitives together".

Indeed in filament/src/HwRenderPrimitiveFactory.h I see this:
```
    // the std::set<T> node size is sizeof(std::set<T>::node_type) + sizeof(T), aligned.
    // we need this value for our pool allocator. On clang 64 bits this is 56 bytes.
```

The assert that fails is "assert(size <= ELEMENT_SIZE);" in libs\utils\include\utils\Allocator.h, with size = 56 and I suspect ELEMENT_SIZE is 48 (and not 56). Isn't this related to the size of DummySetNode ? I'm not sure I understand all the template code yet.

Callstack:
```
utils::PoolAllocator<48,8,0,utils::FreeList>::alloc(unsigned __int64 size, unsigned __int64 alignment, unsigned __int64 offset) Line 332	C++
utils::Arena<utils::PoolAllocator<48,8,0,utils::FreeList>,utils::LockingPolicy::NoLock,utils::TrackingPolicy::HighWatermark,utils::AreaPolicy::HeapArea>::alloc(unsigned __int64 size, unsigned __int64 alignment, unsigned __int64 extra) Line 588	C++
>	utils::STLAllocator<std::_Tree_node<filament::HwRenderPrimitiveFactory::Entry,void *>,utils::Arena<utils::PoolAllocator<48,8,0,utils::FreeList>,utils::LockingPolicy::NoLock,utils::TrackingPolicy::HighWatermark,utils::AreaPolicy::HeapArea>>::allocate(unsigned __int64 n) Line 804	C++
[External Code]	
filament::HwRenderPrimitiveFactory::HwRenderPrimitiveFactory() Line 66	C++
filament::FRenderableManager::FRenderableManager(filament::FEngine & engine) Line 311	C++
```

**Comment by teub** [comment]
I cannot reproduce the assert problem if I only use the shipped libraries; however the crash in release is still here:

```
utils::StructureOfArraysBase<class utils::Arena<class utils::HeapAllocator,struct utils::LockingPolicy::NoLock,struct utils::TrackingPolicy::Untracked,class utils::AreaPolicy::HeapArea>,class utils::EntityInstance<class filament::RenderableManager,0>,class filament::math::details::TMat44<float>,struct filament::FRenderableManager::Visibility,struct filament::FRenderableManager::SkinningBindingInfo,struct filament::FRenderableManager::MorphingBindingInfo,class filament::math::details::TVec3<float>,unsigned short,unsigned char,unsigned short,unsigned char,class filament::math::details::TVec3<float>,class utils::Slice<class filament::FRenderPrimitive,unsigned int>,unsigned int,struct filament::PerRenderableData,float>::forEach<class `utils::StructureOfArraysBase<class utils::Arena<class utils::HeapAllocator,struct utils::LockingPolicy::NoLock,struct utils::TrackingPolicy::Untracked,class utils::AreaPolicy::HeapArea>,class utils::EntityInstance<class filament::RenderableManager,0>,class filament::math::details::()	Unknown
filament::FView::prepare(class filament::FEngine &,class filament::backend::CommandStream &,class utils::ArenaScope<class utils::Arena<class utils::LinearAllocator,struct utils::LockingPolicy::NoLock,struct utils::TrackingPolicy::Untracked,class utils::AreaPolicy::HeapArea> > &,class filament::Viewport const &,struct filament::CameraInfo const &,class filament::math::details::TVec4<float> const &,bool)	Unknown
filament::FRenderer::renderJob(class utils::ArenaScope<class utils::Arena<class utils::LinearAllocator,struct utils::LockingPolicy::NoLock,struct utils::TrackingPolicy::Untracked,class utils::AreaPolicy::HeapArea> > &,class filament::FView &)	Unknown
filament::FRenderer::renderInternal(class filament::FView const *)	Unknown
filament::FRenderer::render(class filament::FView const *)	Unknown
```

Any suggestion or hint welcome !

**Comment by bejado** [comment]
This should be fixed by 39332a92ed330ee1e27f5f5ffefb66b1953e1f43, which went out with the 1.23.2 release. I'll have to look more into this.

**Comment by teub** [comment]
Yes that fixes the assert. I guess I'll probably need to open a new issue for the crash in FScene::prepare. From what I remember the crash happened already in 1.23.1 and I could work around it with some #pragma optimize("", off). I'll have to investigate more.
